cache/main.tf

```diff
@@ -0,0 +1,46 @@
+
+#----------------------------
+# create GCS bucket for cache
+#-----------------------------
+
+resource "google_storage_bucket" "cache" {
+  name          = var.bucket_name
+  location      = var.bucket_location
+  force_destroy = true
+  storage_class = var.bucket_storage_class
+
+  versioning {
+    enabled = var.bucket_versioning
+  }
+
+  uniform_bucket_level_access = true
+
+  lifecycle_rule {
+    condition {
+      age = var.bucket_expiration_days
+    }
+    action {
+      type = "Delete"
+    }
+  }
+  labels = var.bucket_labels
+}
+
+#----------------------------------------------------------------
+# create service account with access to the above created bucket
+#----------------------------------------------------------------
+
+resource "random_id" "suffix" {
+  byte_length = 4
+}
+
+resource "google_service_account" "cache_admin" {
```

Comment:
To be on the more structural side, let's not create this service account. However, let's allow passing the service account email that will have access to the cache bucket. This way, we don't monopolize permissions.